### PR TITLE
fix: avatar

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Avatar/constants.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { ChainId } from '@masknet/web3-shared-evm'
 export const NFT_AVATAR_JSON_SERVER = 'https://configuration.r2d2.to/com.maskbook.avatar.json'
 export const NFT_AVATAR_DB_NAME = 'com.maskbook.user'
 export const NFT_AVATAR_DB_NAME_STORAGE = 'com.maskbook.user.storage'
+export const NFT_AVATAR_METADATA_STORAGE = 'com.maskbook.avatar.metada.storage'
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Avatar/constants.ts]:
```suggestion
export const NFT_AVATAR_METADATA_STORAGE = 'com.maskbook.avatar.metadata.storage'
```

File: packages/mask/src/plugins/Avatar/Application/NFTListDialog.tsx [hunk 2]
```diff
@@ -75,59 +88,73 @@ const useStyles = makeStyles()((theme) => ({
...
         if (!selectedToken?.metadata?.imageURL) return
         setDisabled(true)
+
         try {
-            const image = await downloadUrl(selectedToken.metadata.imageURL)
-            onSelected({ image: URL.createObjectURL(image), account: selectedAccount, token: selectedToken })
-            onNext()
+            const image = await toPNG(selectedToken.metadata.imageURL)
+            if (!image) {
+                showSnackbar('Download image error', { variant: 'error' })
```
Comment (Lanttcat) [inline packages/mask/src/plugins/Avatar/Application/NFTListDialog.tsx]: Maybe should use i18n.

File: packages/mask/src/plugins/Avatar/Application/NFTListDialog.tsx [hunk 3]
```diff
@@ -143,29 +170,19 @@ export function NFTListDialog(props: NFTListDialogProps) {
 
     useEffect(() => setSelectedAccount(account || wallets?.[0]?.identity || ''), [account, wallets])
 
-    const onAddClick = (token: NonFungibleToken<ChainId, SchemaType>) => {
+    const onAddClick = (token: AllChainsNonFungibleToken) => {
+        // TODO reference tokensInList below
```
Comment (Lanttcat) [inline packages/mask/src/plugins/Avatar/Application/NFTListDialog.tsx]: Should remove this?

File: packages/mask/src/plugins/Avatar/Application/NFTListDialog.tsx [hunk 4]
```diff
@@ -200,9 +218,14 @@ export function NFTListDialog(props: NFTListDialogProps) {
...
+    const tokensInList = uniqBy(
+        [...tokens, ...collectibles],
+        selectedPluginId === NetworkPluginID.PLUGIN_SOLANA ? (x) => x.tokenId : (x) => x.contract?.address && x.tokenId,
+    )
+
     const NoNFTList = () => {
-        if (currentPage === POLYGON_PAGE && tokens.length === 0) return AddCollectible
-        else if (currentPage === POLYGON_PAGE && tokens.length) return
+        if (chainId === ChainId.Matic && tokensInList.length === 0) return AddCollectible
+        else if (chainId === ChainId.Matic && tokensInList.length) return
```
Comment (Lanttcat) [inline packages/mask/src/plugins/Avatar/Application/NFTListDialog.tsx]: use if instead? `if (chainId === ChainId.Matic && tokensInList.length) `

File: packages/plugins/Solana/src/state/Hub/hub.ts [hunk 5]
```diff
@@ -66,12 +66,12 @@ class Hub implements SolanaHub {
     }
     getNonFungibleAssets(
         account: string,
-        options?: HubOptions<ChainId> | undefined,
+        options?: HubOptions<ChainId>,
     ): Promise<Pageable<NonFungibleAsset<ChainId, SchemaType>>> {
         try {
-            return SolanaRPC.getNonFungibleAssets(account, options)
-        } catch {
             return MagicEden.getTokens(account, options)
```
Comment (Lanttcat) [inline packages/plugins/Solana/src/state/Hub/hub.ts:72]: Use MagicEden first?

File: packages/mask/src/plugins/Avatar/utils/index.ts [hunk 6]
```diff
@@ -96,6 +96,7 @@ export async function getNFTAvatarByUserId(userId: string, avatarId: string): Pr
             userId.toLowerCase(),
             PLUGIN_ID,
         )
+        console.log(response.val)
```
Comment (Lanttcat) [inline packages/mask/src/plugins/Avatar/utils/index.ts]: Remove this please.

File: packages/plugins/EVM/src/state/Hub/hub.ts [hunk 7]
```diff
@@ -103,8 +103,10 @@ class Hub implements EVM_Hub {
     ): Promise<NonFungibleAsset<ChainId, SchemaType> | undefined> {
         const provider = options?.sourceType
         switch (provider) {
-            case SourceType.OpenSea:
+            case SourceType.OpenSea: {
+                if (options?.chainId !== ChainId.Mainnet) return
```
Comment (Lanttcat) [inline packages/plugins/EVM/src/state/Hub/hub.ts]: Maybe `option.chainId` is not set, and use this.chainId

File: packages/plugins/Solana/src/state/Hub/hub.ts (hunk 5 above)
Comment (lelenei) [reply packages/plugins/Solana/src/state/Hub/hub.ts:72]: solanaRPC is error

File: packages/mask/src/plugins/Avatar/Application/WalletList.tsx [hunk 8]
```diff
@@ -176,41 +179,31 @@ export function AddressNames(props: AddressNamesProps) {
...
-                        wallets.some((x) => isSameAddress(x.identity, account))
-                            ? true
-                            : currentPluginId === NetworkPluginID.PLUGIN_EVM,
+                        wallets.some((x) => isSameAddress(x.identity, account)) ||
+                            currentPluginId === NetworkPluginID.PLUGIN_EVM,
+                        onConnectWallet,
                     )
                 ) : (
-                    <MenuItem key="connect">
+                    <MenuItem key={uuid()}>
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/Application/WalletList.tsx]: Do you mean to use a different key each time?

File: packages/mask/src/plugins/Avatar/SNSAdaptor/AddNFT.tsx [hunk 9]
```diff
@@ -60,21 +64,38 @@ export function AddNFT(props: AddNFTProps) {
...
+                contract: asset.contract,
+                metadata: asset.metadata,
+                tokenId: asset.tokenId,
+                collection: asset.collection,
+            } as AllChainsNonFungibleToken
+        } else {
+            token = await connection.getNonFungibleToken(address, tokenId)
+        }
+        if (!token) {
+            setMessage('cannot found asset')
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/AddNFT.tsx]: Use an i18n key like others.

File: packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarClip.tsx [hunk 10]
```diff
@@ -145,9 +148,13 @@ export function NFTAvatarClip(props: NFTAvatarClipProps) {
     const { id = uuid(), width, height, viewBoxHeight = ViewBoxHeight, viewBoxWidth = ViewBoxWidth, screenName } = props
     const classes = useStylesExtends(useStyles(), props)
     const { loading, value: avatarMetadata } = useNFTContainerAtTwitter(screenName ?? '')
+    const account = useAccount()
     const { value = { amount: '0', symbol: 'ETH', name: '', slug: '' }, loading: loadingNFT } = useNFT(
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarClip.tsx:152]: Why the symbol of an NFT token is `ETH`?

Comment (lelenei) [reply packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarClip.tsx:152]: this is twitter nft

Comment (lelenei) [reply packages/mask/src/plugins/Avatar/SNSAdaptor/NFTAvatarClip.tsx:152]: this is twitter nft

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**